buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt

```diff
@@ -93,16 +108,32 @@ open class BuildScanPlugin : Plugin<Project> {
         if (isCiServer) {
             buildScan {
                 tag("CI")
-                tag(System.getenv("TEAMCITY_BUILDCONF_NAME"))
                 link("TeamCity Build", System.getenv("BUILD_URL"))
                 value("Build ID", System.getenv("BUILD_ID"))
                 setCommitId(System.getenv("BUILD_VCS_NUMBER"))
+
+                whenEnvIsSet("TEAMCITY_BUILDCONF_NAME") { ciBuildConfiguration ->
+                    value(ciConfigurationName, ciBuildConfiguration)
+                    link("Build Configuration Scans", customValueSearchUrl(mapOf(ciConfigurationName to ciBuildConfiguration)))
```

Comment:
I think "Search Same Build Configuration" or "Same Build Configuration Scans" might be better because users may not be able to realize this is a link to the search. WDYT?

Reply:
How about `Scans for same build configuration`?

Should we drop one of `TEAMCITY_BUILDCONF_NAME` and `BUILD_TYPE_ID`? Would you know the difference between the two searches?

Reply:
According to [the doc](https://confluence.jetbrains.com/display/TCD18/Predefined+Build+Parameters), `TEAMCITY_BUILDCONF_NAME` is the name, for example `Test Coverage - Platform Java10 Oracle Windows (workers)`, `BUILD_TYPE_ID` is the unique id, for example `Gradle_Check_Platform_Java10_Oracle_Windows_workers`, I think `BUILD_TYPE_ID` would be more suitable and less errorprone. `Scans for same build configuration` is a liiiitle long, but I guess that's fine.

Reply:
I went for `Build Type Scans`. I think the plural `Scans` should make it clear that you get a list. I also renamed `Teamcity CompileAll Build` to `CI CompileAll Scan`.